jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -352,14 +347,27 @@ public Response createSubscriptionsWithAddOnsInternal(final List<BulkSubscriptio
         }
 
         final EntitlementCallCompletionCallback<List<UUID>> callback = new EntitlementCallCompletionCallback<List<UUID>>() {
+
+            private boolean isImmediateOp = true;
+
             @Override
             public List<UUID> doOperation(final CallContext ctx) throws EntitlementApiException {
+                for (final BaseEntitlementWithAddOnsSpecifier spec: baseEntitlementWithAddOnsSpecifierList) {
+                    if (spec.getBillingEffectiveDate() != null) {
+                        final LocalDate nowInAccountTimeZone = new LocalDate(callContext.getCreatedDate(), spec.getBillingEffectiveDate().getChronology().getZone());
+                        if (spec.getBillingEffectiveDate().isAfter(nowInAccountTimeZone)) {
+                            isImmediateOp = false;
```

Comment:
I am questioning whether this logic will lead to the right expectation:
* `isImmediateOp` = `false` => this was obtained by checking that `spec#getBillingEffectiveDate` is not null and is  in the future, all good.
* `isImmediateOp` = `true` => obtained as a default for when prev condition is not met. However, prev condition only checked that `spec#getBillingEffectiveDate` is not null and is  in the future, but we could still have `spec.getBillingEffectiveDate().is**Not**After(nowInAccountTimeZone)` and yet have a date that is in the future ?

Reply:
Good catch! See 60f3dfe3028ba99abc4859c33380042102e1e7cb.

The only downside is that I had to re-use some of the logic internal to `InternalCallContextFactory` inside jaxrs.